01/lib/raii_action.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+#include <concepts>
+
+
+template <typename T>
+concept callable =
+    requires(T func)
+    {
+        func();
+    };
+
+template <callable T>
+class RAIIAction {
+private:
+    T func;
```

Comment:
Приватному полю класса стоит дать имя с `_` в конце или `m_` в начале 
```suggestion
    T func_
```